File: code/library/src/main/java/org/fbme/lib/st/statements/CaseStatement.java
package org.fbme.lib.st.statements;

import org.fbme.lib.st.expressions.Expression;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface CaseStatement extends Statement {

    @Nullable Expression getExpression();

    void setExpression(@Nullable Expression expression);

    @NotNull List<CaseElement> getCases();

    @Nullable List<Statement> getElseCase();

    @NotNull List<Statement> addElseCase();

    void removeElseCase();
}

File: code/library/src/main/kotlin/org/fbme/lib/st/statements/CaseStatement.kt
package org.fbme.lib.st.statements

import org.fbme.lib.st.expressions.Expression

interface CaseStatement : Statement {
    var expression: Expression?
    val cases: MutableList<CaseElement>
    val elseCase: List<Statement>?
    fun addElseCase(): MutableList<Statement>
    fun removeElseCase()
}
